bigtable_rs/src/bigtable.rs

```diff
@@ -258,13 +258,13 @@ impl BigTableConnection {
     /// https://docs.rs/tokio/latest/tokio/attr.main.html, but it might be a very different case for
     /// different applications.
     ///
-    pub fn new_with_auth_manager(
+    pub fn new_with_token_provider(
         project_id: &str,
         instance_name: &str,
         is_read_only: bool,
         channel_size: usize,
         timeout: Option<Duration>,
-        authentication_manager: AuthenticationManager,
+        token_provider: Arc<dyn TokenProvider>,
```

Comment:
I could also become a bit more generic here, so we don't need to `Arc::new` on the call site, but i leave it up to you to decide.